src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2831,47 +2831,55 @@ public void visitTypeCast(JCTypeCast tree) {
      */
     public void visitTypeTest(JCInstanceOf tree) {
         if (tree.expr.type.isPrimitive() || tree.pattern.type.isPrimitive()) {
-            JCExpression exactnessCheck = null;
+            JCStatement sideEffect;
```

Comment:
As a mostly stylistic comment: if I were doing this, I would create a variable like: `List<JCStatement> statements;` (or any other name), and then would assign either `List.nil()` or `List.of(...)` to it. That way there would be no `null` checks later.